src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenRecorder.cs

```diff
@@ -40,18 +40,21 @@ public class ScreenRecorder : MonoBehaviour
 
         private int _frameCountSinceLastTick;
 
-        private string _currentVideoDirectory;
+        private string _currentGameplaySessionDirectoryPrefix;
 
         private CancellationTokenSource _tokenSource;
 
         private static ScreenRecorder _this;
+        
+        private static RGServiceManager _rgServiceManager = new();
```

Comment:
@RG-nAmKcAz I tried to grab an existing RGServiceManager in Awake, but it was not created by the time this object was awake. Would you prefer this approach, or one that allows this behaviour to create the singleton?

Reply:
NEVER.. (mostly).. create a monobehaviour using 'new'.. there are very rare exceptions but this case isn't one of them.

We have discussed the topic of unity event functions in the past, but accessing other objects in Awake is unstable as there is no guarantee the other objects are Awake yet.  If you have something you want to happen any time the object wakes back up, you can use OnEnable, or if you want it to happen once and only once, you can use Start.

https://docs.unity3d.com/Manual/ExecutionOrder.html

Reply:
(discussed in Discord, this seems correct)